apps/frontend/src/components/global/Sidebar.vue

```diff
@@ -90,36 +203,92 @@ const SidebarProps = Vue.extend({
   components: {
     LinkItem,
     FileItem,
+    ProfileItem,
     AboutModal,
     HelpModal
   }
 })
 export default class Sidebar extends SidebarProps {
-  //selects/deselects all files. Will select all unless all are already selected
-  toggle_all(): void {
-    if (this.all_toggled) {
-      FilteredDataModule.set_toggled_files([]);
+  // used to toggle v-exapansion-panel
+  file_views = 0;
+
+  created() {
+    // open the appropriate v-expansion-panel based on current route
+    if (this.curr_route_path == '/results') this.file_views = 0;
+    else if (this.curr_route_path == '/compare') this.file_views = 0;
+    else if (this.curr_route_path == '/profiles') this.file_views = 1;
+  }
+
+  // select the specified files
+  set_selected_files(files: number[]): void {
+    FilteredDataModule.set_toggled_files(files);
+  }
+
+  // toggle the "select all" for profiles
+  toggle_all_profiles(): void {
+    if (this.all_toggled_profiles) {
+      this.set_selected_files(this.selected_evaluations);
+    } else {
+      let files = InspecDataModule.allProfileFiles.map(v => v.unique_id);
+      files.push(...this.selected_evaluations);
+      this.set_selected_files(files);
+    }
+  }
+
+  // check to see if all profiles is selected
+  get all_toggled_profiles(): boolean {
+    return (
+      this.selected_profiles.length == InspecDataModule.allProfileFiles.length
+    );
+  }
+
+  // toggle the "select all" for evaluations
+  toggle_all_evaluations(): void {
+    if (this.all_toggled_evaluation) {
+      this.set_selected_files(this.selected_profiles);
     } else {
-      FilteredDataModule.set_toggled_files(
-        InspecDataModule.allFiles.map(v => v.unique_id)
-      );
+      let files = InspecDataModule.allEvaluationFiles.map(v => v.unique_id);
+      files.push(...this.selected_profiles);
+      this.set_selected_files(files);
     }
   }
 
-  //checks if all files are selected
-  get all_toggled(): boolean {
+  // check to see if all evalutions are selected
+  get all_toggled_evaluation(): boolean {
     return (
-      InspecDataModule.allFiles.length ==
-      FilteredDataModule.selected_file_ids.length
+      this.selected_evaluations.length ==
+      InspecDataModule.allEvaluationFiles.length
```

Comment:
If this is replaced with `FilteredDataModule.selected_evaluations` then the `selected_evaluations` computed property can be removed from Sidebar.vue.